### PR TITLE
server/node: add cluster id mismatch reason in error

### PR DESCRIPTION
**Pull request by nolouch**
PTAL @siddontang @disksing

### CODE REVIEW

File: src/server/node.rs [hunk 1]
```diff
@@ -169,7 +169,8 @@ impl<C> Node<C>
 
         let ident = res.unwrap();
         if ident.get_cluster_id() != self.cluster_id {
-            error!("cluster ID mismatch: local_id {} remote_id {}",
+            error!("cluster ID mismatch: local_id {} remote_id {}. \
+                    maybe you try to connect another cluster, please reconnect to the correct PD",
```
Comment (siddontang) [inline src/server/node.rs]: remove "maybe", here makes sure connecting to a wrong cluster.

**Comment by siddontang** [comment]
PTAL @disksing

File: src/server/node.rs [hunk 2]
```diff
@@ -169,7 +169,8 @@ impl<C> Node<C>
 
         let ident = res.unwrap();
         if ident.get_cluster_id() != self.cluster_id {
-            error!("cluster ID mismatch: local_id {} remote_id {}",
+            error!("cluster ID mismatch: local_id {} remote_id {}. \
+                    you try to connect another cluster, please reconnect to the correct PD",
```
Comment (andelf) [inline src/server/node.rs]: are trying 
connect to

**Comment by nolouch** [comment]
PTAL @andelf

**Comment by disksing** [comment]
LGTM.

**Pull request merged**